src/main/java/htsjdk/tribble/bed/BEDCodec.java

```diff
@@ -197,7 +201,13 @@ private void createExons(int start, String[] tokens, FullBEDFeature gene,
 
     @Override
     public boolean canDecode(final String path) {
-        return path.toLowerCase().endsWith(".bed");
+        final String toDecode;
+        if (AbstractFeatureReader.hasBlockCompressedExtension(path)) {
+            toDecode = path.substring(0, path.lastIndexOf("."));
+        } else {
+            toDecode = path;
+        }
+        return toDecode.toLowerCase().endsWith(".bed");
```

Comment:
Sure. Do you prefer a simple `String` constant or a `List<String>` for the case of new extensions are added later?